YOLO/ViewController.swift

```diff
@@ -502,78 +662,47 @@ class ViewController: UIViewController {
+        } else {
+          if task == .detect {
+            let offset = (ratio - 1) * (0.5 - displayRect.maxY)
+            let transform = CGAffineTransform(scaleX: 1, y: -1).translatedBy(x: 0, y: offset - 1)
+            displayRect = displayRect.applying(transform)
+          } else {
+            let offset = (ratio - 1) * (0.5 - displayRect.minY)
+            let transform = CGAffineTransform(translationX: 0, y: offset)
+            displayRect = displayRect.applying(transform)
+          }
           ratio = (height / width) / (3.0 / 4.0)
-          rect.size.height /= ratio
+          displayRect.size.height /= ratio
         }
+        displayRect = VNImageRectForNormalizedRect(displayRect, Int(width), Int(height))
 
-        // Scale normalized to pixels [375, 812] [width, height]
-        rect = VNImageRectForNormalizedRect(rect, Int(width), Int(height))
-
-        // The labels array is a list of VNClassificationObservation objects,
-        // with the highest scoring class first in the list.
-        let bestClass = prediction.labels[0].identifier
-        let confidence = prediction.labels[0].confidence
-        // print(confidence, rect)  // debug (confidence, xywh) with xywh origin top left (pixels)
-        let label = String(format: "%@ %.1f", bestClass, confidence * 100)
-        let alpha = CGFloat((confidence - 0.2) / (1.0 - 0.2) * 0.9)
-        // Show the bounding box.
         boundingBoxViews[i].show(
-          frame: rect,
-          label: label,
-          color: colors[bestClass] ?? UIColor.white,
-          alpha: alpha)  // alpha 0 (transparent) to 1 (opaque) for conf threshold 0.2 to 1.0)
+          frame: displayRect, label: label, color: boxColor, alpha: alpha)
 
         if developerMode {
-          // Write
           if save_detections {
             str += String(
               format: "%.3f %.3f %.3f %@ %.2f %.1f %.1f %.1f %.1f\n",
               sec_day, freeSpace(), UIDevice.current.batteryLevel, bestClass, confidence,
               rect.origin.x, rect.origin.y, rect.size.width, rect.size.height)
           }
-
-          // Action trigger upon detection
-          // if false {
-          //     if (bestClass == "car") {  // "cell phone", "car", "person"
-          //         self.takePhoto(nil)
-          //         // self.pauseButton(nil)
-          //         sleep(2)
-          //     }
-          // }
         }
+
       } else {
         boundingBoxViews[i].hide()
       }
     }
-
-    // Write
-    if developerMode {
-      if save_detections {
-        saveText(text: str, file: "detections.txt")  // Write stats for each detection
-      }
-      if save_frames {
-        str = String(
-          format: "%.3f %.3f %.3f %.3f %.1f %.1f %.1f\n",
-          sec_day, freeSpace(), memoryUsage(), UIDevice.current.batteryLevel,
-          self.t1 * 1000, self.t2 * 1000, 1 / self.t4)
-        saveText(text: str, file: "frames.txt")  // Write stats for each image
-      }
-    }
```

Comment:
@john-rocky Why did you remove this logging?

Reply:
@ambitious-octopus Sorry, it is my mistake. I reverted it.